### ISSUE TITLE
Failing to initiate activity adapter

### ISSUE THREAD

**Issue by niyobobo** [guenodz/livedata-recyclerview-sample#1]
Why is this line showing an error and failed to update adapter. Does access the getAllpost without observing to the livedata providing erros. because to access getallpost by observing not getvalue and I saw the result but when using getvalue it shows NPE for the List that is passing into the constructor of an adapter

**Error is on this line:**
`postsAdapter = new PostsAdapter(postViewModel.getAllPosts().getValue(), this, this);`

while setting adapter it carry an non-object reference to adapter's constructor and fail to initiate datasource of recycleview .

`java.lang.NullPointerException: Attempt to invoke interface method 'int java.util.List.size()' on a null object reference`

**Comment by guenodz** [comment]
Hi @niyobobo 
There is a mistake in this code, I will fix it ASAP. This is due to an exception when calling `getValue()` this happens because the getValue() method is trying to get unavailable data. According to LiveData docs, the data (in a live data object) will be available only one a listener is attached to the live data object via `observe()` method.

**Comment by niyobobo** [comment]
I tried to acces live data by obserrving to the getAllpost method but the problem I am facing is when will `setdata()` method in adapter be used? because setting data through adapter constructor and notify the adapter that the data have been changed by `adapter.notifyDataSetChanged()` makes whole recycleview to be updated whenever a new item added/inserted.
What i am trying to say is how will `DiffUtil` be used in this case.

Looking forward to your update.
